oidcserver/handlers.go

```diff
@@ -704,10 +704,14 @@ func (s *Server) handleRefreshToken(w http.ResponseWriter, r *http.Request, clie
 	// this interface can't perform refreshing.
 	if refreshConn, ok := conn.(RefreshConnector); ok {
 		newIdent, err := refreshConn.Refresh(r.Context(), parseScopes(scopes), ident)
-		if err != nil {
+		if isRetryableErr(err) {
 			s.logger.Errorf("failed to refresh identity: %v", err)
 			s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
 			return
+		} else if err != nil {
+			s.logger.Errorf("failed to refresh identity: %v", err)
+			s.tokenErrHelper(w, errInvalidGrant, "", http.StatusBadRequest)
+			return
```

Comment:
what if there is a general server error, that could be retried?

Reply:
Yah, that's the intent here. HTTP 500 can be considered retryable, whereas HTTP 400 with `invalid_grant` is not.

Reply:
nevermind me, i read in the wrong order